src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,59 @@
+package duke;
+
+import java.util.StringJoiner;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+
+    private LocalDateTime by;
+    private DateTimeFormatter inputFormat = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+    private DateTimeFormatter outputFormat = DateTimeFormatter.ofPattern("d MMMM Y hh:mma");
+
+    /**
+     * Duke.Duke.Deadline Constructor.
+     * @param description Description of deadline.
+     * @param by Date string formatted in the form "d/M/yyyy HHmm".
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = LocalDateTime.parse(by, inputFormat);
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by.format(outputFormat) + ")";
+    }
+
+    /**
+     * Used in saving of task data for storage.
+     * @return Formatted string representation of deadline.
+     */
+    @Override
+    public String getSaveString() {
```

Comment:
Good use of the StringJoiner class here. Code looks neat and is good use when we need a fixed format for saving into a file.